rest/blip_sync_messages.go

```diff
@@ -0,0 +1,229 @@
+	sinceSequenceId := s.sinceZeroValueCreator()
+
+	if sinceStr, found := s.rq.Properties["since"]; found {
+		var err error
+		if sinceSequenceId, err = db.ParseSequenceIDFromJSON([]byte(sinceStr)); err != nil {
+			s.logger.LogTo("Sync", "%s: Invalid sequence ID in 'since': %s", s.rq, sinceStr)
+			sinceSequenceId = db.SequenceID{}
+		}
+	}
+
+	return sinceSequenceId
+
+}
+
+func (s *subChanges) batchSize() int {
+	return int(getRestrictedIntFromString(s.rq.Properties["batch"], BlipDefaultBatchSize, 10, math.MaxUint64, true))
+}
+
+func (s *subChanges) continuous() bool {
+	continuous := false
+	if val, found := s.rq.Properties["continuous"]; found && val != "false" {
+		continuous = true
+	}
+	return continuous
+}
+
+func (s *subChanges) activeOnly() bool {
+	return (s.rq.Properties["active_only"] == "true")
+}
+
+func (s *subChanges) filter() string {
+	return s.rq.Properties["filter"]
+}
+
+func (s *subChanges) channels() (channels string, found bool) {
+	channels, found = s.rq.Properties["channels"]
+	return channels, found
+}
+
+func (s *subChanges) channelsExpandedSet() (resultChannels base.Set, err error) {
+	channelsParam, found := s.rq.Properties["channels"]
+	if !found {
+		return nil, fmt.Errorf("Missing 'channels' filter parameter")
+	}
+	channelsArray := strings.Split(channelsParam, ",")
+	return channels.SetFromArray(channelsArray, channels.ExpandStar)
+}
+
+// Satisfy fmt.Stringer interface for dumping attributes of this subChanges request to logs
+func (s *subChanges) String() string {
+
+	buffer := bytes.NewBufferString("")
+
+	buffer.WriteString(fmt.Sprintf("Since:%v ", s.since()))
+
+	continuous := s.continuous()
+	if continuous {
+		buffer.WriteString(fmt.Sprintf("Continuous:%v ", continuous))
+	}
+
+	activeOnly := s.activeOnly()
+	if activeOnly {
+		buffer.WriteString(fmt.Sprintf("ActiveOnly:%v ", activeOnly))
+	}
+
+	filter := s.filter()
+	if len(filter) > 0 {
+		buffer.WriteString(fmt.Sprintf("Filter:%v ", filter))
+		channels, found := s.channels()
+		if found {
+			buffer.WriteString(fmt.Sprintf("Channels:%v ", channels))
+		}
+	}
+
+	batchSize := s.batchSize()
+	if batchSize != int(BlipDefaultBatchSize) {
+		buffer.WriteString(fmt.Sprintf("BatchSize:%v ", s.batchSize()))
+	}
+
+	return buffer.String()
+
+}
+
+type setCheckpoint struct {
+	rq *blip.Message // The underlying BLIP message
+}
+
+func newSetCheckpoint(rq *blip.Message) *setCheckpoint {
+	return &setCheckpoint{
+		rq: rq,
+	}
+}
+
+func (s *setCheckpoint) client() string {
+	return s.rq.Properties["client"]
+}
+
+func (s *setCheckpoint) rev() string {
+	return s.rq.Properties["rev"]
+}
+
+func (s *setCheckpoint) String() string {
+
+	buffer := bytes.NewBufferString("")
+
+	buffer.WriteString(fmt.Sprintf("Client:%v ", s.client()))
+
+	rev := s.rev()
+	if len(rev) > 0 {
+		buffer.WriteString(fmt.Sprintf("Rev:%v ", rev))
+	}
+
+	return buffer.String()
+
+}
+
+type addRevision struct {
+	rq *blip.Message // The underlying BLIP message
+}
```

Comment:
As above - it feels like the helpers are in a bit of a middle ground, where they aren't quite doing enough to justify their existence.

If they were getting rid of some of the hard-to-parse code in the handlers themselves (like the `found && del != "0" && del != "false"` mentioned in an earlier comment), then I think there could be value.  

If parsing the request was computationally expensive, and the helpers were ensuring that the parsing only happened once, that might be useful.  

As they currently stand, they feel like they are just adding a layer of indirection that makes it harder to see what the calling code is doing with the request.